gusto_embedded/src/ReactSDKProvider.tsx

```diff
@@ -1,24 +1,51 @@
+import { HTTPClient } from "./lib/http.js";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import { GustoEmbeddedProvider } from "./react-query/index.js";
 import { GustoEmbeddedCore } from "./core.js";
 
 //Reexporting useQueryClient to be available in consumer application without direct react-query dependency
-export { useQueryClient } from "@tanstack/react-query"
+export { useQueryClient } from "@tanstack/react-query";
 
-export function ReactSDKProvider({ url, children }: { url: string, children: React.ReactNode }) {
-    const queryClient = new QueryClient();
-    const gustoClient = new GustoEmbeddedCore({
-        serverURL: url,
-    },)
-    queryClient.setQueryDefaults(['@gusto/embedded-api'], { retry: false })
-    queryClient.setMutationDefaults(['@gusto/embedded-api'], { retry: false })
+export function ReactSDKProvider({
+  url,
+  headers,
+  children,
+}: {
+  url: string;
+  headers?: Record<string, string | number>;
+  children: React.ReactNode;
+}) {
+  const httpClientWithHeaders = new HTTPClient({
+    fetcher: async (request) => {
+      if (request instanceof Request) {
```

Comment:
What other type could `request` have?

Reply:
`Fetcher` is defined [here](https://github.com/Gusto/gusto-typescript-client/blob/dfd17c7131b217d9bb604e71c0f5d398836a268b/gusto_embedded/src/lib/http.ts#L5-L8) with the argument type `RequestInfo | URL`, and `RequestInfo` is defined as `type RequestInfo = string | Request` (I can't find a link to any reference, but this was what my IDE showed me). The only time the request would have explicit headers defined is when the actual type is `Request`.